From e191dc1d0dd8086067ac4a241d51501bd8f93060 Mon Sep 17 00:00:00 2001
From: Dusty Mabe <dusty@dustymabe.com>
Date: Mon, 27 Nov 2023 15:53:13 -0500
Subject: [PATCH 5/5] tools/osbuild-mpp: set part ID from name if missing

For our Fedora CoreOS disk images we set the partition labels (name)
for the partitions. This is also supported using the primitives here
in OSBuild, but it wasn't obvious that I needed to set the name in
the mpp-define-images definition. Let's set the name there, but let's
also allow osbuild-mpp to set the `id`, which is what is used later
to access that partition from the `name` too if `id` isn't set.

This means we allow something like:

   - name: BIOS-BOOT
     type: 21686148-6449-6E6F-744E-656564454649
     bootable: true
     uuid: FAC7F1FB-3E8D-4137-A512-961DE09A5549
     size: 100

rather than requiring something like:

   - id: BIOS-BOOT
     name: BIOS-BOOT
     type: 21686148-6449-6E6F-744E-656564454649
     bootable: true
     uuid: FAC7F1FB-3E8D-4137-A512-961DE09A5549
     size: 100
---
 tools/osbuild-mpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/osbuild-mpp b/tools/osbuild-mpp
index c2f440a7..3214e647 100755
--- a/tools/osbuild-mpp
+++ b/tools/osbuild-mpp
@@ -283,6 +283,7 @@ Example:
           "partitions": [
             {
               "id": "bios-boot",
+              "name": "BIOS-BOOT",
               "start": 2048,
               "size": 2048,
               "type": "21686148-6449-6E6F-744E-656564454649",
@@ -820,7 +821,8 @@ class Partition:
                  name: str = None,
                  uuid: str = None,
                  attrs: List[int] = None):
-        self.id = uid
+        # if no id provided, use the part-label (name) as the id
+        self.id = uid or name
         self.type = pttype
         self.start = start
         self.size = size
-- 
2.41.0

